app/src/main/java/app/suhasdissa/vibeyou/ui/screens/player/FullScreenPlayer.kt

```
@@ -118,13 +118,24 @@ fun FullScreenPlayer(
                 contentDescription = stringResource(id = R.string.album_art),
                 contentScale = ContentScale.Crop
             ) {
-                val state = painter.state
-                if (state is AsyncImagePainter.State.Loading || state is AsyncImagePainter.State.Error) {
-                    Image(
-                        painter = rememberAsyncImagePainter(model = it.mediaMetadata.artworkUri),
+                val state1 = painter.state
+                if (state1 !is AsyncImagePainter.State.Success) {
+                    SubcomposeAsyncImage(
+                        model = it.mediaMetadata.artworkUri,
                         contentDescription,
                         contentScale = contentScale
-                    )
+                    ) {
+                        val state2 = painter.state
```

`state1` and `state2` would always be equal because they're reacting to changes to `painter.state` I think? And this looks way too nested to me, I'd prefer to make it simpler if possible.

---

No, the state1 is for the outer(high res) image
And the state2 is for the inner(low res) image

The `painter` os actually a member of the SubcomposeAsyncImageScope

This first tries to load the high res one, if it fails, this will try loading the low res image.
If all fails, this will fall back to the placeholder image.

---

Oh, I see ...
I thought the painter was introduced somewhere above.
Still, I'd prefer a solution with less nesting, e.g. if we had a boolean `val isError by remember { mutableStateOf(false) }` that we set to true if there's an error and then do
```kotlin
if (isError) {
   Image(//placeholder)
}
```
In the same if else statement, so that it's less nested and easier to read.